src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.ts

```diff
@@ -940,6 +947,19 @@ export class ProgrammingExerciseUpdateComponent implements OnInit {
         }
     }
 
+    private createProgrammingExerciseForImportFromFile() {
+        this.programmingExercise = history.state.programmingExerciseForImportFromFile;
+        this.programmingExercise.id = undefined;
+        this.programmingExercise.projectKey = undefined;
+        this.programmingExercise.dueDate = undefined;
+        this.programmingExercise.assessmentDueDate = undefined;
+        this.programmingExercise.releaseDate = undefined;
+        this.programmingExercise.startDate = undefined;
+        //without dates set, they can only be false
+        this.programmingExercise.allowComplaintsForAutomaticAssessments = false;
+        this.programmingExercise.allowManualFeedbackRequests = false;
+    }
```

Comment:
What is actually kept here?
Would it be better to create a new programming exercise and copying the values that you want to keep?

Reply:
allowOnlineEditor, staticCodeAnalysisEnabled, allowOfflineIde, sequentialTestruns, showTestNamesToStudents, testwiseCoverageEnabled, projectType, programmingLanguage, packageName and the zip file that belongs to the import
So, I think it doesn't really matter which way you choose.
For me, this one was more intuitive because I explicitly reset the values and don't need to think about any values I might miss to reassign to the new object.

Reply:
What about adding new properties later on? Is it better to explicitly have to exclude or include them in here?